server/src/services/documentSymbol/SymbolVisitor.ts

```diff
@@ -27,20 +28,18 @@ export abstract class SymbolVisitor {
     // Find identifier
     const childCursor = cursor.spawn();
 
-    do {
-      const nameToken: TokenNode | null = childCursor.findTokenWithKind([
-        this.nameTokenKind,
-      ]);
+    while (childCursor.goToNextTokenWithKind(this.nameToken[1])) {
+      if (childCursor.nodeName === this.nameToken[0]) {
```

Comment:
nit suggestion: short circuit to reduce nesting:

```suggestion
      if (childCursor.nodeName !== this.nameToken[0]) {
        continue;
      }
```

Reply:
Done in https://github.com/NomicFoundation/hardhat-vscode/pull/545/commits/9f279fd80fc66a049308cb83c7b380372a2f343c